zebra/zapi_msg.c

```diff
@@ -3182,16 +3185,19 @@ static void zread_vrf_label(ZAPI_HANDLER_ARGS)
 			}
 		}
 
-		if (really_remove)
+		if (really_remove) {
 			mpls_lsp_uninstall(def_zvrf, ltype, zvrf->label[afi],
 					   NEXTHOP_TYPE_IFINDEX, NULL,
 					   ifp->ifindex, false /*backup*/);
+			mpls_auto_interface_data_off(ifp);
+		}
 	}
 
 	if (nlabel != MPLS_LABEL_NONE) {
 		mpls_label_t out_label = MPLS_LABEL_IMPLICIT_NULL;
 		mpls_lsp_install(def_zvrf, ltype, nlabel, 1, &out_label,
 				 NEXTHOP_TYPE_IFINDEX, NULL, ifp->ifindex);
+		mpls_auto_interface_data_on(ifp);
```

Comment:
Shouldn't this be turned on before installing MPLS LSP?

Reply:
no, this should be turned on, only when used.
actually, setting the /proc/sys/net/mpls/input flag may have negative impact on kernel, when interface flapping happens, and when platform_auto value has huge value. 
this method avoids generalising the usage of the input flag on every interface.